src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembershipInternal.java

```diff
@@ -3,26 +3,89 @@
+import uk.gov.companieshouse.acsp.manage.users.service.AcspMembersService;
+import uk.gov.companieshouse.acsp.manage.users.service.UsersService;
+import uk.gov.companieshouse.acsp.manage.users.utils.StaticPropertyUtil;
+import uk.gov.companieshouse.api.accounts.user.model.UsersList;
 import uk.gov.companieshouse.api.acsp_manage_users.api.UserAcspMembershipInternalInterface;
 import uk.gov.companieshouse.api.acsp_manage_users.model.InternalRequestBodyPatch;
 import uk.gov.companieshouse.api.acsp_manage_users.model.InternalRequestBodyPost;
 import uk.gov.companieshouse.api.acsp_manage_users.model.ResponseBodyPost;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
 
 @RestController
 public class UserAcspMembershipInternal implements UserAcspMembershipInternalInterface {
 
-  public UserAcspMembershipInternal() {
-    // Empty constructor as yet to inject any dependencies
+  private static final Logger LOG =
+      LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
+  private final AcspDataService acspDataService;
+  private final UsersService usersService;
+  private final AcspMembersService acspMembersService;
+
+  @Autowired
+  public UserAcspMembershipInternal(
+      final AcspDataService acspDataService,
+      final UsersService usersService,
+      final AcspMembersService acspMembersService) {
+    this.acspDataService = acspDataService;
+    this.usersService = usersService;
+    this.acspMembersService = acspMembersService;
   }
 
   @Override
   public ResponseEntity<ResponseBodyPost> addAcspOwner(
-      @NotNull String xRequestId,
-      @Pattern(regexp = "^[0-9A-Za-z-_]{0,32}$") String acspNumber,
-      @Valid InternalRequestBodyPost internalRequestBodyPost) {
-    return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1238)
+      final String xRequestId,
+      final String acspNumber,
+      final InternalRequestBodyPost internalRequestBodyPost) {
+    LOG.debugContext(
+        xRequestId,
+        String.format("Attempting to add ACSP owner for acsp_number %s", acspNumber),
+        null);
+    acspDataService.fetchAcspData(acspNumber);
+    final String userEmail = internalRequestBodyPost.getOwnerEmail();
+    final UsersList users =
+        Optional.ofNullable(usersService.searchUserDetails(List.of(userEmail)))
+            .filter((userList -> !userList.isEmpty()))
+            .orElseThrow(
+                () -> {
+                  LOG.error(String.format("Failed to find user with userEmail %s", userEmail));
+                  return new NotFoundRuntimeException(
+                      StaticPropertyUtil.APPLICATION_NAMESPACE, "Failed to find user");
+                });
+    final String userId = users.getFirst().getUserId();
+    final var acspMembers = acspMembersService.fetchAcspMemberships(users.getFirst(), false);
+    if (!acspMembers.isEmpty()) {
```

Comment:
dont have to filter, isNotEmpty hear means association exists, so we dont have to do extra checks,
we can throw error straight away.
if(!acspMembers.isEmpty()){throw error}

Reply:
No, that's not true. This will return all associations for the user, but it doesn't mean it's the one for the particular acsp number.